pkg/config/config.go

```diff
@@ -78,8 +78,18 @@ type KubernetesConfig struct {
 
 // APISIXConfig contains all APISIX related config items.
 type APISIXConfig struct {
-	BaseURL string `json:"base_url" yaml:"base_url"`
+	// DefaultClusterName is the name of default cluster.
+	DefaultClusterName string `json:"default_cluster_name"`
+	// DefaultClusterBaseURL is the base url configuration for the default cluster.
+	DefaultClusterBaseURL string `json:"default_cluster_base_url" yaml:"default_cluster_base_url"`
+	// DefaultClusterAdminKey is the admin key for the default cluster.
 	// TODO: Obsolete the plain way to specify admin_key, which is insecure.
+	DefaultClusterAdminKey string `json:"default_cluster_admin_key" yaml:"default_cluster_admin_key"`
+	// BaseURL is same to DefaultClusterBaseURL.
+	// Deprecated: use DefaultClusterBaseURL instead.
+	BaseURL string `json:"base_url" yaml:"base_url"`
```

Comment:
Should we have a deprecation policy?  e.g. when to actually remove it.

Reply:
What about:

```
// Deprecated: use DefaultClusterBaseURL instead, BaseURL will be removed once v1.0.0 is released.
```

Reply:
SGTM.

Reply:
Changed.

Reply:
Changed.